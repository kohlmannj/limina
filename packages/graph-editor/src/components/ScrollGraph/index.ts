export { default } from './ScrollGraph';
export * from './ScrollGraph';
