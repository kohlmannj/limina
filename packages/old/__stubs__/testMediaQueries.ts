export default [320, 540, 854, 1280, 1605];
