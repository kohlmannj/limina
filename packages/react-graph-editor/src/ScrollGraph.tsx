import React, { Component, CSSProperties, SFC } from 'react';
import styled, { css } from 'react-emotion';
import { withProps } from 'recompose';
import Graph, { IGraphProps } from './Graph';

export interface IScrollGraphProps extends IGraphProps {
  scaleX?: number;
  scaleY?: number;
  style?: CSSProperties;
}

const HorizontalRange = styled('input')`
  width: 150px;
  height: 15px;
  margin: 0;
  position: absolute;
  left: 0;
  bottom: 0;
  z-index: 1;
  transition: 0.25s opacity ease-in-out;
  opacity: 0;

  &:hover,
  &:focus {
    opacity: 1;
  }
`;

const VerticalRange = styled(HorizontalRange)`
  transform-origin: 7.5px 50%;
  transform: rotate(90deg);
  top: 0;
  right: -135px;
  bottom: auto;
  left: auto;
`;

const GraphContainer = styled.div`
  width: 100%;
  height: 100%;
  overflow: auto;
  background: rgba(0, 0, 0, 0.25);
`;

const GraphTrack = styled.div`
  box-sizing: border-box;
  padding-right: 15px;
  padding-bottom: 15px;
`;

const ScrollGraph: SFC<IScrollGraphProps> = ({ className, scaleX, scaleY, style, ...rest }) => (
  <GraphContainer className={className} style={style}>
    <GraphTrack
      className={css({
        width: `${scaleX! * 100}%`,
        height: `${scaleY! * 100}%`,
      })}
    >
      <Graph {...rest} />
    </GraphTrack>
    <HorizontalRange type="range" value={scaleX} min={1} max={10} />
    <VerticalRange type="range" value={scaleY} min={1} max={10} />
  </GraphContainer>
);

ScrollGraph.defaultProps = {
  scaleX: 1,
  scaleY: 1,
};

export default ScrollGraph;
