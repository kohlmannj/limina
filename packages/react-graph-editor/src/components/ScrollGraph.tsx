import React, { CSSProperties, SFC } from 'react';
import styled, { css } from 'react-emotion';
import { withProps } from 'recompose';
import Graph, { IGraphProps } from './Graph';
import ScrollView from './ScrollView';

export interface IScrollGraphProps extends IGraphProps {
  scaleX?: number;
  scaleY?: number;
  style?: CSSProperties;
}

const GraphContainer = styled.div`
  overflow: auto;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.25);
`;

const GraphTrack = styled.div`
  box-sizing: border-box;
  padding-right: 15px;
  padding-bottom: 15px;
`;

const ScrollGraph: SFC<IScrollGraphProps> = ({ className, scaleX, scaleY, style, ...rest }) => (
  <ScrollView className={className} style={style}>
    <Graph
      className={css`
        width: ${(scaleX || 1) * 100}%;
        height: ${(scaleY || 1) * 100}%;
      `}
      {...rest}
    />
  </ScrollView>
);

ScrollGraph.defaultProps = {
  scaleX: 1,
  scaleY: 1,
};

export default ScrollGraph;
