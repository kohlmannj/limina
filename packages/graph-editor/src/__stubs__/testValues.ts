import { ILineProps } from '../components/Line';
import { PointTuple } from '../index.d';

export const testPoints: PointTuple[] = [[320, 320], [480, 0], [1280, 1160]];

export const testLines: ILineProps[] = [
  { label: 'fontSize', points: [[320, 12], [540, 16], [1280, 24]] },
  { label: 'width', points: [[320, 320], [540, 480], [1280, 1160]] },
  { label: 'height', points: [[320, 320], [540, 405], [1280, 652.5]] },
];
