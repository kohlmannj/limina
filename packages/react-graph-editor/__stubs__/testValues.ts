import { PointTuple } from '../src';
import { IPolyLineSVG } from '../src/PolyLineSVG';

export const testPoints: PointTuple[] = [[320, 320], [480, 0], [1280, 1160]];

export const testLines: IPolyLineSVG[] = [
  { label: 'fontSize', points: [[320, 12], [540, 16], [1280, 24]] },
  { label: 'width', points: [[320, 320], [540, 480], [1280, 1160]] },
  { label: 'height', points: [[320, 320], [540, 405], [1280, 652.5]] },
];
