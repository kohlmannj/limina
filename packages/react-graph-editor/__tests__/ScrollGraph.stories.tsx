import { storiesOf } from '@storybook/react'; // tslint:disable-line no-implicit-dependencies
import React from 'react';
import { testLines } from '../__stubs__/testValues';
import ScrollGraph from '../src/ScrollGraph';

storiesOf('react-graph-editor/ScrollGraph', module)
  .add('with default scales', () => (
    <ScrollGraph style={{ width: '100%', height: '100vh' }} lines={testLines} />
  ))
  .add('with scaleX = 2', () => (
    <ScrollGraph style={{ width: '100%', height: '100vh' }} scaleX={2} lines={testLines} />
  ))
  .add('with scaleX = 10', () => (
    <ScrollGraph style={{ width: '100%', height: '100vh' }} scaleX={10} lines={testLines} />
  ))
  .add('with scaleY = 2', () => (
    <ScrollGraph style={{ width: '100%', height: '100vh' }} scaleY={2} lines={testLines} />
  ))
  .add('with scaleY = 10', () => (
    <ScrollGraph style={{ width: '100%', height: '100vh' }} scaleY={10} lines={testLines} />
  ))
  .add('with scaleX = scaleY = 2', () => (
    <ScrollGraph
      style={{ width: '100%', height: '100vh' }}
      scaleX={2}
      scaleY={2}
      lines={testLines}
    />
  ))
  .add('with scaleX = scaleY = 10', () => (
    <ScrollGraph
      style={{ width: '100%', height: '100vh' }}
      scaleX={10}
      scaleY={10}
      lines={testLines}
    />
  ));
