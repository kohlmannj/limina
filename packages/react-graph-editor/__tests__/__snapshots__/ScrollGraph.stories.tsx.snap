// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-graph-editor/ScrollGraph with default scales 1`] = `
.emotion-0 {
  width: 100%;
  height: 100%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={1}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={1}
  />
</GraphContainer>
`;

exports[`react-graph-editor/ScrollGraph with scaleX = 2 1`] = `
.emotion-0 {
  width: 200%;
  height: 100%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={2}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={1}
  />
</GraphContainer>
`;

exports[`react-graph-editor/ScrollGraph with scaleX = 10 1`] = `
.emotion-0 {
  width: 1000%;
  height: 100%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={10}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={1}
  />
</GraphContainer>
`;

exports[`react-graph-editor/ScrollGraph with scaleX = scaleY = 2 1`] = `
.emotion-0 {
  width: 200%;
  height: 200%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={2}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={2}
  />
</GraphContainer>
`;

exports[`react-graph-editor/ScrollGraph with scaleX = scaleY = 10 1`] = `
.emotion-0 {
  width: 1000%;
  height: 1000%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={10}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={10}
  />
</GraphContainer>
`;

exports[`react-graph-editor/ScrollGraph with scaleY = 2 1`] = `
.emotion-0 {
  width: 100%;
  height: 200%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={1}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={2}
  />
</GraphContainer>
`;

exports[`react-graph-editor/ScrollGraph with scaleY = 10 1`] = `
.emotion-0 {
  width: 100%;
  height: 1000%;
}

<GraphContainer
  style={
    Object {
      "height": "100vh",
      "width": "100%",
    }
  }
>
  <GraphTrack
    className="emotion-0"
  >
    <StyledGraphContainer
      lines={
        Array [
          Object {
            "label": "fontSize",
            "points": Array [
              Array [
                320,
                12,
              ],
              Array [
                540,
                16,
              ],
              Array [
                1280,
                24,
              ],
            ],
          },
          Object {
            "label": "width",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                480,
              ],
              Array [
                1280,
                1160,
              ],
            ],
          },
          Object {
            "label": "height",
            "points": Array [
              Array [
                320,
                320,
              ],
              Array [
                540,
                405,
              ],
              Array [
                1280,
                652.5,
              ],
            ],
          },
        ]
      }
    />
  </GraphTrack>
  <HorizontalRange
    max={10}
    min={1}
    type="range"
    value={1}
  />
  <VerticalRange
    max={10}
    min={1}
    type="range"
    value={10}
  />
</GraphContainer>
`;
